### ISSUE TITLE
Icons for file with only one extension

### ISSUE THREAD

**Issue by theloadstar** [athityakumar/colorls#469]
- Relevant Issues : (none)
- Relevant PRs : (none)
- Type of issue :
  - [ ] Installation
  - [x] Font-related
  - [ ] Feature request
  - [ ] Bug in existing feature
  - [ ] Developer mode : Code quality / Tests / Documentation

Hello, the colorls in my machine can't recognize file with only one extension such as `.ignore`, `.DS_Store`,`.type_example`, here is an example:
<img width="368" alt="image" src="https://user-images.githubusercontent.com/48984008/128820448-916ae612-0624-4067-9a83-029960d51df5.png">

The file `.DS_Store` and `.gitignore` are recognized as `file` type instead of `apple` / `git` type. I changed nothing in `yaml` dir.(I also tried adding `ds_store:"\ue711"`, and it didn't work.)

If I add any latters before dot, it can recognize rightly. For example, I renamed `.gitignore` to `test.gitignore`,it showed as follow:
<img width="370" alt="image" src="https://user-images.githubusercontent.com/48984008/128820519-09125616-0c0d-4112-b5e1-ac98518a1aa2.png">

Other files with name before dot can be recognized correctly. My system is macOS 10.15.7, and I have installed nerd-font.

Thank you for reading.

**Comment by avdv** [comment]
Thanks for the bug report!

The problem is that the files you mentioned do not have an extension. 

For example, `.gitignore` is a hidden file, which is denoted by the leading dot, but by there is no file extension.

We would need to change the implementation that we no longer recognise files by extension, but rather using a suffix match, such that if one wanted to specify a icon for a file with an extension you would have to include the dot in the pattern.

**Comment by theloadstar** [comment]
Thank you for replying.

I tried adding `.DS_Store:         apple` in the file `file_aliases.yaml` before and didn't work, wondering how to realize `if one wanted to specify an icon for a file with an extension you would have to include the dot in the pattern`. 

I noticed in `README.md`, the colorls can recognize the files that do not have an extension like `.gitignore` and `.DS_Store`, I just wondering how does it work, do I need to change something in `file_aliases.yaml` or `file.yaml`?
<img width="908" alt="image" src="https://user-images.githubusercontent.com/48984008/129172915-6373f80a-4b82-4719-a749-4cdb8b371d0a.png">

My English is not so good, thank you for reading :)

**Comment by avdv** [comment]
You are right, it used to work in older versions, but I broke it in commit https://github.com/athityakumar/colorls/commit/17b2142c48ac6cd760532116c0e483085caaea2c while fixing #141. 

(In my defence, there were no unit tests assuring this works the way it did)

**Comment by theloadstar** [comment]
Got it, thank you for replying :)

**Comment by avdv** [comment]
Why did you close this? This is a valid issue that is supposed to work!

**Comment by theloadstar** [comment]
Sorry, I thought after the commit 17b2142c48ac6cd760532116c0e483085caaea2c colorls won't bring this feature back again, my mistake , ಠ_ಠ

**Comment by olirwin** [comment]
Hi, would a code snippet like this [here](https://github.com/athityakumar/colorls/blob/1a5a0c9cd0ef9a9c122929c72c1a8fb8c205063e/lib/colorls/core.rb#L384) :

```ruby
extension = File.extname(content.name).delete_prefix(".")
basename = File.basename(content.name, ".*").delete_prefix(".")
key = extension.empty? ? basename : extension
key = key.downcase.to_sym
```

solve this issue ? This might also solve #459 
I have never done any ruby before so this might not be optimised / up to standard etc, but it seems to work on online compiler environments
